subprojects/docs/src/samples/userguide/files/copy/kotlin/build.gradle.kts

```diff
@@ -150,7 +151,7 @@ task<Copy>("anotherCopyTask") {
     // Copy the contents of a Zip file
     from(zipTree("src/main/assets.zip"))
     // Determine the destination directory later
-    into { getDestDir() }
+    into(Callable { getDestDir() })
```

Comment:
Did this actually compile/work before? If so, how?

Also, this API should really be made more friendly from the Kotlin DSL: #1636

Being able to just do `into { getDestDir() }` would be really nice.

Reply:
It did compile but fail at runtime saying that function doesn't support the provider construct, a kotlin function. See https://github.com/gradle/kotlin-dsl/issues/1077
Won't fix in this PR.